### PR TITLE
Use knative eventing event receiver and CE client

### LINKED ISSUE #15: Propagate traces through channel
The CloudEvents Distributed Tracing Extension must be used to propagate traces through the Kafka Channel.  ~~Traces must be written to the location specified in config-tracing~~ _This aspect will be addressed in a later story after controller refactoring_

See the following requirement:
https://github.com/knative/eventing/blob/master/docs/spec/channel.md#generic

### PR DESCRIPTION
**Pull request by sbgreene1307**
Switch to using the knative-eventing cloud event client and the knative-eventing event_receiver. Together, these services provide the implementation that conforms to the Channel Specification around trace header propagation and creation of the traceparent and tracestate headers 

#15

### CODE REVIEW

File: components/channel/cmd/main.go [hunk 1]
```diff
@@ -40,42 +40,44 @@ func main() {
...
+	// internally, that client goes unused when using the ServeHTTP
+	// on the eventReceiver.
+	// IMPORTANT: Because the kncloudevents package does not allow
+	// injecting modified configuration, we can't override the
+	// default port being used, 8080.
+	knCloudEventClient, err := kncloudevents.NewDefaultClient()
 	if err != nil {
-		logger.Error("failed to create client", zap.Error(err))
+		logger.Error("failed to knative cloud event create client", zap.Error(err))
 		return
```
Comment (travis-minke-sap) [inline components/channel/cmd/main.go:81]: grammer  ; )

File: components/channel/cmd/main.go [hunk 2]
```diff
@@ -40,42 +40,44 @@ func main() {
...
+	// The Knative CloudEvent Client handles the mux http server
+	// setup (middlewares and transport options) and invokes
+	// the eventReceiver. Althought the NewEventReceiver method
+	// above will also invoke kncloudevents.NewDefaultClient
+	// internally, that client goes unused when using the ServeHTTP
+	// on the eventReceiver.
+	// IMPORTANT: Because the kncloudevents package does not allow
+	// injecting modified configuration, we can't override the
+	// default port being used, 8080.
+	knCloudEventClient, err := kncloudevents.NewDefaultClient()
```
Comment (travis-minke-sap) [inline components/channel/cmd/main.go:78]: Assuming there are changes later on changing the channel service port configuration in the controller?   (Yep I found them)

**Review by phamilton — APPROVED** [review]
LGTM

**Pull request merged**